### ISSUE TITLE
Condition Editor

### ISSUE THREAD

**Issue by moiri** [humdek-unibe-ch/sh-selfhelp#463]
The naming in the condition editor is confusing:

When selecting the `Field` the condition can be of the form `[Field Name]` <operator> `[value]`.

I would have expected the `[Field Name]` to be the actual name of the field to which a `[value]` is compared.
However, if I use the field `my_id` which is an actual field existing for each record and `{{p_record_id}}` which is replaced by the ID of the parent record to which I want the `my_id` field value to be compared, the condition fails.

Of course, I can use `{{my_id}}` as `[Field Name]` but this results in a value to value comparison.
This works but it is not what I would expect when reading the labels.

There are now three questions/notes in this issue:
- [ ] confusing naming
- [ ] comparison of zero-filled strings and numbers
- [ ] filtering on `entryList` not working

**Comment by moiri** [comment]
Comparing IDs with the `=` operator is messy because IDs are sometimes represented as zero-filled strings and sometimes as numbers.

**Comment by moiri** [comment]
Why is the condition not working on the `entryList` but only on the children.
Shouldn't the filtering happen **before** the items are all rendered?

**Comment by Kodzhabashev** [comment]
I just tested and the condition works on `entryList`. Can you give me an example. What is the condition? Is the condition loading variables form another entry or form itself?

**Comment by moiri** [comment]
Yes, in general the conditions work but not on the record fields because they are only available in the children of the entry list.
